client/core/core.go

```diff
@@ -1687,12 +1705,28 @@ func (c *Core) NewDepositAddress(assetID uint32) (string, error) {
 	dbWallet.Address = addr
 	err = c.db.UpdateWallet(dbWallet)
 	if err != nil {
-		return "", fmt.Errorf("UpdateWallet error for %s: %w", unbip(assetID), err)
+		return "", fmt.Errorf("UpdateWallet error for %s: %w", unbip(w.AssetID), err)
 	}
+	// Update wallet state on user struct
+	walletState := w.state()
+	c.setUserWalletState(walletState)
+	c.notify(newWalletStateNote(walletState))
 
-	c.refreshUser()
+	return addr, nil
+}
 
-	c.notify(newWalletStateNote(w.state()))
+// NewDepositAddress retrieves a new deposit address from the specified asset's
+// wallet and saves it to the database.
+func (c *Core) NewDepositAddress(assetID uint32) (string, error) {
+	w, exists := c.wallet(assetID)
+	if !exists {
+		return "", newError(missingWalletErr, "no wallet found for %s", unbip(assetID))
+	}
+
+	addr, err := c.newDepositAddress(w)
+	if err != nil {
+		return "", err
+	}
 
 	return addr, nil
```

Comment:
Since we've defined `newDepositAddress` above to return an empty string for a non-nil error, it would be acceptable in this case to just do `return c.newDepositAddress(w)`, although generally it's wise to ensure return values work this way when dealing with exported methods from other packages.

Anyway, I've not actually tested any of this PR, so feel free to hold off for further comments.

Reply:
Right, can do